app/scripts/modules/core/src/pipeline/config/pipelineConfigurer.js

```diff
@@ -537,5 +543,19 @@ module.exports = angular
       if ($scope.isTemplatedPipeline && $scope.pipeline.isNew && !$scope.hasDynamicSource) {
         this.configureTemplate();
       }
+
+      this.saveTemplate = template => {
+        return PipelineTemplateWriter.savePipelineTemplateV2(template).then(
+          response => {
+            console.log('response', response);
+            const id = response.variables.find(v => v.key === 'pipelineTemplate.id').value;
+            $state.go('home.pipeline-templates.pipeline-templates-detail', {
+              templateId: PipelineTemplateV2Service.idForTemplate({ id }),
+            });
+            return true;
+          },
+          err => err,
```

Comment:
I think when this errors out, this func will cause the promise to appear as fulfilled to the `then` in `ShowPipelineTemplateJsonModal` line 64 and close the modal without surfacing that something went wrong.

Reply:
I've added a throw here to propagate to the catch() handler of the modal.